pkg/operator/pdbcontroller/pdb_controller.go

```diff
@@ -0,0 +1,147 @@
+package pdbcontroller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configv1listers "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+const (
+	infrastructureClusterName = "cluster"
+)
+
+// Controller ensures that the guarding workload and pdb are available and running
+type Controller struct {
+	operatorClient       v1helpers.OperatorClient
+	kubeClient           kubernetes.Interface
+	infrastructureLister configv1listers.InfrastructureLister
+	clusterTopology      configv1.TopologyMode
+	targetNamespace      string
+	pdbManifest          []byte
+	factory              *factory.Factory
+	eventRecorder        events.Recorder
+}
+
+var _ factory.Controller = &Controller{}
+
+func New(operatorClient v1helpers.OperatorClient,
```

Comment:
I'm ok with this controller as a short term solution if @sttts and  @soltysh want to land in 4.8.  If they prefer waiting until 4.9, please develop the idea of a precondition which can decorate a static resource controller.

`NewStaticResourceController().WithConditionalResource(conditionFunc, manifest, deleteConditionFunc)`.  If the deleteConditionFunc returns true, delete the resource, if the conditionFunc returns true create the resource. Deletion trumps creation.  By default, deletionCondtionFunc is !conditionFunc.